model_signing/serialization/dfs.py

```diff
@@ -188,18 +205,25 @@ def __init__(
 
     @override
     def serialize(self, model_path: pathlib.Path) -> manifest.Manifest:
-        # TODO: #196 - Add checks to exclude symlinks if desired
-        _check_file_or_directory(model_path)
+        # Note: This function currently uses `pathlib.Path.glob` so the DFS
```

Comment:
I think we use `sorted()` to sort https://github.com/sigstore/model-transparency/blob/db894e203611eb35027314464d66991c66bfecfc/model_signing/serialization/dfs.py#L129 and https://github.com/sigstore/model-transparency/blob/db894e203611eb35027314464d66991c66bfecfc/model_signing/serialization/dfs.py#L224 so I don't think we rely on the glob implementation?

Reply:
Yes, we sort the output. But we still need to treat files and directories afterwards, whereas `Path.walk` will give them to us directly and in a guaranteed DFS traversal no matter what.

The difference would arrive when we want to ignore a directory. With `glob`, we have to analyze any element in the returned list and remove if directory is a parent. With `walk`, as soon as the directory is found, we cut the search short and continue from the next entry.